modules/mongodb/collect.go

```diff
@@ -109,3 +109,73 @@ func (m *Mongo) collectDbStats(ms map[string]int64) error {
 	}
 	return nil
 }
+
+// replSetCollect creates the map[string]int64 for the available dims.
+// nil values will be ignored and not added to the map and thus metrics
+// should not appear on the dashboard.
+// if the querying node does not belong to a replica set
+func (m *Mongo) collectReplSetStatus(ms map[string]int64) error {
+	status, err := m.mongoCollector.replSetGetStatus()
+	if err != nil {
+		return fmt.Errorf("error get server status from mongo: %s", err)
+	}
+	var currentMembers []string
+	for _, member := range status.Members {
+		currentMembers = append(currentMembers, member.Name)
+	}
+
+	// replica nodes may be removed
+	// we should collect metrics for these anymore
+	m.removeReplicaSetMembers(currentMembers)
+	m.replSetMembers = currentMembers
+
+	for _, member := range status.Members {
+		// Heartbeat lag calculation
+		if member.LastHeartbeatReceived != nil {
+			id := replicationHeartbeatLatencyDimPrefix + member.Name
+			// add dimension if not exists yet
+			if !m.replSetDimsEnabled[id] {
+				m.replSetDimsEnabled[id] = true
+				chart := m.charts.Get(replicationHeartbeatLatency)
+				if chart != nil {
+					if err := chart.AddDim(&module.Dim{ID: id, Name: member.Name}); err != nil {
+						m.Warningf("failed to add dim with id: %s, err: %v", id, err)
+					}
+				}
+			}
+			ms[id] = status.Date.Sub(*member.LastHeartbeatReceived).Milliseconds()
+		}
+
+		// Replica set time diff between current time and time when last entry from the oplog was applied
+		id := replicationLagDimPrefix + member.Name
+		// add dimension if not exists yet
+		if !m.replSetDimsEnabled[id] {
+			m.replSetDimsEnabled[id] = true
+			chart := m.charts.Get(replicationLag)
+			if chart != nil {
+				if err := chart.AddDim(&module.Dim{ID: id, Name: member.Name}); err != nil {
+					m.Warningf("failed to add dim with id: %s, err: %v", id, err)
+				}
+			}
+		}
+		ms[id] = status.Date.Sub(member.OptimeDate).Milliseconds()
+
+		// Ping time
+		if member.PingMs != nil {
+			id := replicationNodePingDimPrefix + member.Name
+			// add dimension if not exists yet
+			if !m.replSetDimsEnabled[id] {
+				m.replSetDimsEnabled[id] = true
+				chart := m.charts.Get(replicationNodePing)
+				if chart != nil {
+					if err := chart.AddDim(&module.Dim{ID: id, Name: member.Name}); err != nil {
+						m.Warningf("failed to add dim with id: %s, err: %v", id, err)
```

Comment:
```suggestion
					if err := chart.AddDim(&module.Dim{ID: id, Name: member.Name}); err != nil {
						m.Warningf("failed to add dim: %v", err)
```

The error contains both the chart and dim ids

Reply:
done